hashintel/hash
# Using the wait and start timeouts from env by default for tests

## 🌟 What is the purpose of this PR?

The testing suite had values the start and wait timeouts for the Engine hard-coded. This is fine for normal operation, such as running the full test suite, but it'd be good to allow manual overrides by the user in case of debugging a long-running test, or trying to get a faster feedback loop.

We also previously had a wait timeout (the length of time the CLI waits for an update from the engine before assuming it's stalled) of 600 seconds which was far too long and made us think the test suite ran infinitely in some cases.

## ⚠️ Known issues

The CLI will still wait for the wait-timeout length of time even if the Engine process has crashed.

## 🐾 Next steps

Fix the crash issue explained above.

## 🔍 What does this change?

- Checks for `ENGINE_WAIT_TIMEOUT` and `ENGINE_START_TIMEOUT` from the environment before setting them to a default value
- Makes the `ExperimentConfig` one function up in the integration tests to avoid passing as many parameters
- Changes the default wait timeout to 60 seconds instead of 600

### 📜 Does this require a change to the docs?

- No

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201793759102797/1201757298172584/f) (_internal_)
- [Follow-up task for when the engine crashes](https://app.asana.com/0/1199548034582004/1201837823804295/f)(_internal_)

## 🛡 What tests cover this?

- The integration test suite (checked by the CI)

## ❓ How to test this?

1.  Pick some test-case where the Engine crashes (check one of the ignored tests in Python for example)
2. Run the test, see that it takes a long time to finish (~60 seconds)
3. Set `ENGINE_WAIT_TIMEOUT` to 10 seconds, check that it finishes a lot faster
